app/code/Magento/Theme/Controller/Result/JsFooterPlugin.php

```diff
@@ -39,26 +39,38 @@ public function __construct(ScopeConfigInterface $scopeConfig)
      */
     public function beforeSendResponse(Http $subject)
     {
-        $content = $subject->getContent();
-        $script = [];
-        if (is_string($content) && strpos($content, '</body') !== false) {
-            if ($this->scopeConfig->isSetFlag(
-                self::XML_PATH_DEV_MOVE_JS_TO_BOTTOM,
-                ScopeInterface::SCOPE_STORE
-            )
-            ) {
-                $pattern = '#<script[^>]*+(?<!text/x-magento-template.)>.*?</script>#is';
-                $content = preg_replace_callback(
-                    $pattern,
-                    function ($matchPart) use (&$script) {
-                        $script[] = $matchPart[0];
-                        return '';
-                    },
-                    $content
-                );
-                $subject->setContent(
-                    str_replace('</body', implode("\n", $script) . "\n</body", $content)
-                );
+        $content = $subject->getContent() ?? '';
```

Comment:
I think instead of null coalesce here we should cast this as a string:

`$content = (string) $subject->getContent();`